src/ssh/cockpitsshrelay.c

```diff
@@ -982,19 +974,21 @@ cockpit_ssh_authenticate (CockpitSshData *data)
         {
           auth_func = do_interactive_auth;
           method = SSH_AUTH_METHOD_INTERACTIVE;
-          has_creds = data->initial_auth_data != NULL && \
-                      (g_strcmp0 (data->auth_options->auth_type, "basic") == 0 ||
-                       g_strcmp0 (data->auth_options->auth_type,
-                                 auth_method_description (method)) == 0);
+          has_creds = data->in_bridge ||
```

Comment:
I wonder if `in_bridge` is really sufficient? As far as I can see, `data->initial_auth_data` is set in lines 687 and 748 when called through the bridge, but `ssh_askpass()` can fail, and return `NULL`. I suppose if it does then we'll just get an auth failure later and there's nothing that we can do anyway, as we don't do "basic" auth through the bridge? It's just a bit confusing to get `has_creds == TRUE` here even if that's not so.